Mathlib/Data/List/SplitLengths.lean

```diff
@@ -0,0 +1,80 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.Data.List.Join
+import Mathlib.Order.MinMax
+import Mathlib.Algebra.BigOperators.Group.List
+
+/-!
+# Splitting a list to chunks of specified lengths
+
+This file defines splitting a list to chunks of given lengths, and some proofs about that.
+-/
+
+variable {α : Type*} (l : List α) (sz : List ℕ)
+
+/--
+Split a list to chunks of given lengths.
+-/
+def List.splitLengths : List α → List ℕ → List (List α)
```

Comment:
I would suggest swapping the arguments. Given a list of lengths this is then a nice operator `List a \to List (List a)`. Also the fact that's the way the induction runs!